server/lib/nl/detection/place.py

```diff
@@ -48,93 +48,56 @@ def detect_from_query_dc(orig_query: str,
-        if 'dcid' in rp:
-          related.append(rp['dcid'])
-      main2corrections[main] = related
-
-      # For logging, get all DCIDs:
-      debug_logs["dc_recognize_places"][item['span']] = [
-          d['dcid'] for d in item['places'] if 'dcid' in d
-      ]
-    else:
-      query_parts.append(QueryPart(substr=item['span'].lower()))
-
-  resolved_places = []
-  resolved_entities = []
-  parent_map = {}
-  if mains:
-    resolved_places, parent_map = get_place_from_dcids(
-        mains, debug_logs["place_resolution"])
-    if allow_triples:
-      resolved_entities = _get_non_place_entities(mains, resolved_places)
-  resolved_place_dcids = set([p.dcid for p in resolved_places])
-  resolved_entity_dcids = set([e.dcid for e in resolved_entities])
+  place_query_parts = _get_query_parts(query, debug_logs["dc_recognize_places"],
+                                       True)
+  # Use the first DCID for now.
+  place_main_dcids = [p.dcids[0] for p in place_query_parts if p.dcids]
+  resolved_places, parent_map = get_place_from_dcids(
+      place_main_dcids, debug_logs["place_resolution"])
 
   main_place = None
-  peers = []
   parent_places = []
   if resolved_places:
     main_place = resolved_places[0]
     parent_places = parent_map.get(main_place.dcid, [])
 
-  # Set PlaceDetection.
-  places_str = []
+  resolved_place_dcids = set([p.dcid for p in resolved_places])
+  places_str, stripped_query_str = _get_stripped_string(place_query_parts,
+                                                        resolved_place_dcids)
+
+  resolved_entities = []
   entities_str = []
-  stripped_query_parts = []
-  for p in query_parts:
-    retain_in_stripped_query = True
-    if p.dcid in resolved_entity_dcids:
-      # The query part is an entity dcid so add its original string to entities_str
-      entities_str.append(p.substr)
-      if len(resolved_entity_dcids) > _MAX_ENTITIES_QUERY:
-        # Remove the query part if there are more than _MAX_ENTITIES_QUERY
-        # number of non place entities resolved
-        retain_in_stripped_query = False
-    if p.dcid in resolved_place_dcids:
-      # The query part is a place dcid so add its original string to places_str
-      places_str.append(p.substr)
-      # Always remove the query part if it is a place
-      retain_in_stripped_query = False
-    if retain_in_stripped_query:
-      stripped_query_parts.append(p.substr)
-  query_without_place_substr = ' '.join(stripped_query_parts)
+  if allow_triples:
+    debug_logs["dc_recognize_entities"] = {}
+    entity_query_parts = _get_query_parts(stripped_query_str,
+                                          debug_logs["dc_recognize_entities"],
+                                          False)
+    resolved_entities = _get_resolved_entities(entity_query_parts)
+    resolved_entity_dcids = set([e.dcid for e in resolved_entities])
+    entities_str, stripped_entities_query_str = _get_stripped_string(
```

Comment:
Add vertical spacing, and call out why we pass `stripped_query_str`?

Reply:
done